lib/api/include/irods/ticketAPI.hpp

```diff
@@ -0,0 +1,405 @@
+#ifndef IRODS_TICKET_API_HPP
+#define IRODS_TICKET_API_HPP
+
+#include <string_view>
+#include <string>
+
+#undef NAMESPACE_IMPL
+#undef RxComm
+#undef rxTicketAdmin
+
+#ifdef IRODS_RESOURCE_ADMINISTRATION_ENABLE_SERVER_SIDE_API
+#    define NAMESPACE_IMPL server
+#    define RxComm         RsComm
+#    define rxTicketAdmin  rsTicketAdmin
+
+struct RsComm;
+#else
+#    define NAMESPACE_IMPL client
+#    define RxComm         RcComm
+#    define rxTicketAdmin  rcTicketAdmin
+
+struct RcComm;
+#endif // IRODS_RESOURCE_ADMINISTRATION_ENABLE_SERVER_SIDE_API
+
+enum ticket_type
```

Comment:
Change enum to enum class

Reply:
Put in namespace

Reply:
Yes, the enumerations can go inside of `ticket` namespace, but not in the `client`/`server` namespace.

These enumerations are safe to use across the client and server implementations.

For example ...
```cpp
namespace irods::experimental::administration::ticket
{
    enum class ticket_type;
    enum class ticket_operation;

    namespace NAMESPACE_IMPL
    {
        auto create_type(..., ticket_type t, ...) -> void;
    }
}
```